### DISCUSSION TITLE
Adonis 4.1 - Update many tables at the same time.

### DISCUSSION THREAD

**Discussion by pauloscript** [adonisjs/core#1292] [Help] [answered]
Hi, guys!

I have many 1: N relationships. I know how to update 1 relationship at a time, now I tried to update many relationships at the same time, but it didn't work lol, and I tried it in many ways. Does anyone have an example of how to do this or where can I find it in the documentation?

What I do is: A class has N phones, N addresses and belongs to a city.

So, I don't know how to update all of this at the same time. I hope I have made it clear, I thank everyone who can help or criticize.

**Comment by McSneaky** [comment, marked as answer]
I'm not too sure what you are looking for?
Perhaps something like sync()? https://adonisjs.com/docs/4.1/relationships#_sync
Or just multi update? You can use bulk update for that: https://adonisjs.com/docs/4.0/lucid#_bulk_updates
```js
await ModelToUpdate.query().update({ phone: 123123).where('classAId', 10 });
```

**Reply by pauloscript** [reply]
Hi, I thank you for your attention.

In the documentation I found sync () well explained, with attach() and detach() (I'll be using it soon). And yes, I think the bulk update would be as close to what I want to do now, but I didn't quite understand how it works, because I have a hasMany and belongsTo relationship, how could I update the information in these tables related to another at the same time, passing my data that comes (for example) from a form. 

Would I use Query Builder to get my data (make the query, with a join I think) and then update it in bulk? 
Because in the examples I see, the data is always passed directly in the query with the update, not in a method for example.

Does Adonis not offer any method like merge () for example for this type of mass update? I started researching more about bulk updates to see if I can do it. 


I am learning about Adonis, so I apologize for the question that may be silly or lack of understanding of how it works.
Again I thank you for your attention and answer.

**Reply by McSneaky** [reply]
I'm not sure what you want to update, but let's say you have users and user has many phones. Phone can be disabled or enabled. Disabled means no calls can be made to this phone

Let's say you have objects like this
```js
User {
  id: 5,
  name: 'McSneaky',
}
```
This user has many phones
```js
[
  {
    id: 59,
    name: 'Nokia 3310',
    user_id: 5,
    disabled: false,
    number: 123,
  },
  {
    id: 85,
    name: 'Alcatel OneTouch',
    user_id: 5,
    disabled: false,
    number: 321,
  },
]
```
Now let's say I log in to this platform and want to disable all my phones, since they got stolen
```js
await Phone.query().update({ disabled: true }).where('user_id', auth.user.id)
```
Since I know that hasMany relationship uses `related_id` on other model. In current case `user_id` I can make such query :slightly_smiling_face: 

I hope this example helps lil bit and I didn't miss it with topic too much :sweat_smile:

**Reply by pauloscript** [reply]
This ER represents a little how my entities are. 

- The property belongs to a city (1:1).

- The property has an address (1:1).

- The property can have many telephones (1:N).

![example](https://user-images.githubusercontent.com/59545484/88095164-0c81e500-cb6b-11ea-8e2b-b53cdcaccbf3.PNG)

When I register a property the other entities will be registered... I'm following the documentation and did the store () method, I don't know if that's the correct way to do it, but it worked, when I call the index () or show () method, I can get all the information.

![example_code](https://user-images.githubusercontent.com/59545484/88095770-03454800-cb6c-11ea-862e-1fdbbb55ab07.PNG)

Now I'm trying to do the update () method, but I'm not sure how to do it and I also didn't understand in the documentation how I could be doing this.
So what I'm trying to do is: When I update the property and if I have a phone etc., they will update it together. So, I believe that async () doesn't apply here, maybe the bulk update does. But I didn't understand exactly how it works for everyone for these relationships.

I started doing this, but I believe that the update () would be more wrong than the store () method, lol.
.
.
.
`    property.merge(data)
    await property.save()
   `

`     if(address) {
      await property.address().update(address)
    }`
.
.
.

The `await property.address().update(address)` I think it would work here, but not for the telephone hasMany.

So that's why I'm here, I don't know if the other methods are "correct" (they work, but would it be okay to do that? Lol) and I'm not sure how to do this update (), I need a light haha. I think it’s easier to understand.
Thanks for your help!

**Comment by McSneaky** [comment]
That's a late replay but better than nothing :yum: 
- There's nothing wrong with using `update` method
- I don't think you can update property AND phones in one query, but you can make multiple
  
You already have property update there with merge and save and if property has multiple phones then how are they meant to be updated? If property has 10 phones, then how will you know which one was meant to be updated?
